src/XMakeBuildEngine/Evaluation/ProjectRootElementCache.cs

```diff
@@ -227,6 +227,8 @@ internal ProjectRootElement Get(string projectFile, OpenProjectRootElement openP
                             // use: it checks the file content as well as the timestamp. That's better than completely disabling
                             // the cache as we get test coverage of the rest of the cache code.
                             XmlDocument document = new XmlDocument();
+                            //  TODO: does PreserveWhitespace need to be an option here?
+                            //document.PreserveWhitespace = true;
```

Comment:
I've updated this so that it won't use the cached version if it doesn't match the preserveFormatting setting requested.